Sources/Commands/SwiftBuildTool.swift

```diff
@@ -204,6 +209,40 @@ public struct SwiftBuildTool: SwiftTool {
     
         return try (mode ?? .build(.debug, UserToolchain()), opts)
     }
+
+    private func checkClangVersion() {
+        // We only care about this on Ubuntu 14.04
+        guard let uname = try? popen(["lsb_release", "-r"]).chomp(),
+              uname.hasSuffix("14.04"),
+              let clangVersionOutput = try? popen(["clang", "--version"]).chomp(),
+              let minorVersion = SwiftBuildTool.getClangMinorVersion(versionOutput: clangVersionOutput) else {
+            return
+        }
+
+        if minorVersion < 6 {
+            print("warning: minimum recommended clang is version 3.6, otherwise you may encounter linker errors.")
+        }
+    }
+
+    /// Get clang's minor version from the given version output string on Ubuntu.
+    static func getClangMinorVersion(versionOutput: String) -> Int? {
+        // Clang outputs version in this format on Ubuntu:
+        // Ubuntu clang version 3.6.0-2ubuntu1~trusty1 (tags/RELEASE_360/final) (based on LLVM 3.6.0)
+        let versionStringPrefix = "Ubuntu clang version "
+        let versionStrings = versionOutput.utf8.split(separator: UInt8(ascii: "-")).flatMap(String.init)
+        guard let clangVersionString = versionStrings.first,
+              clangVersionString.hasPrefix(versionStringPrefix) else {
+            return nil
+        }
+        let versionStartIndex = clangVersionString.index(clangVersionString.startIndex, offsetBy: versionStringPrefix.utf8.count)
+        let versionString = clangVersionString[versionStartIndex..<clangVersionString.endIndex]
+        // Split major minor patch etc.
+        let versions = versionString.utf8.split(separator: UInt8(ascii: ".")).flatMap(String.init)
+        guard versions.count > 1 else {
+            return nil
+        }
+        return Int(versions[1])
+    }
```

Comment:
I meant to test it in isolation unrelated to SwiftBuiltTool (like, `Utilities`?)